src/humanloop/evals/run.py

```diff
@@ -493,69 +631,66 @@ def _file_or_file_inside_hl_utility(file: File) -> File:
+    return file_config
 
 
-def _get_file_callable(file: File, type_: FileType) -> Optional[Callable]:
-    # Get the `callable` from the `file` to Evaluate
-    function_ = typing.cast(Optional[Callable], file.pop("callable", None))
+def _get_file_callable(file_config: FileEvalConfig) -> Optional[Callable]:
+    """Get the callable of the File to be evaluated, or throw if None was provided for Flows."""
+    type_ = file_config.get("type")
+    function_ = typing.cast(Optional[Callable], file_config.pop("callable", None))
     if function_ is None:
         if type_ == "flow":
-            raise ValueError("You must provide a `callable` for your Flow `file` to run a local eval.")
+            raise HumanloopRuntimeError("You must provide a `callable` for your Flow `file` to run a local eval.")
         else:
-            sys.stdout.write(
-                f"{CYAN}No `callable` provided for your {type_} file - will attempt to generate logs on Humanloop.\n\n{RESET}"
+            print_info(
+                f"No `callable` provided for your {type_} file - will attempt to generate logs on Humanloop.\n\n"
             )
+    elif type_ == "agent":
+        raise ValueError("Agent evaluation is only possible on the Humanloop runtime, do not provide a `callable`.")
     return function_
 
 
-def _upsert_file(
-    file: File, type: FileType, client: "BaseHumanloop"
-) -> Union[PromptResponse, FlowResponse, ToolResponse, EvaluatorResponse]:
+def _upsert_file(client: "BaseHumanloop", file_config: FileEvalConfig) -> EvaluatedFile:
     # Get or create the file on Humanloop
-    version = file.pop("version", {})
-    file_dict = {**file, **version}
+    version = file_config.pop("version", {})
+    file_dict = {**file_config, **version}
+    del file_dict["type"]
+    type_ = file_config.get("type")
+    subclient = _get_subclient(client=client, file_config=file_config)
 
-    if type == "flow":
+    if type_ == "flow":
         # Be more lenient with Flow versions as they are arbitrary json
         try:
             Flow.model_validate(version)
         except ValidationError:
             flow_version = {"attributes": version}
-            file_dict = {**file, **flow_version}
-        hl_file = client.flows.upsert(**file_dict)  # type: ignore [arg-type, assignment]
-
-    elif type == "prompt":
+            file_dict = {**file_config, **flow_version}
+    elif type_ == "prompt":
         # Will throw error if version is invalid
         Prompt.model_validate(version)
-        hl_file = client.prompts.upsert(**file_dict)  # type: ignore [arg-type, assignment]
-
-    elif type == "tool":
+    elif type_ == "tool":
         # Will throw error if version is invalid
         Tool.model_validate(version)
-        hl_file = client.tools.upsert(**file_dict)  # type: ignore [arg-type, assignment]
-
-    elif type == "evaluator":
-        hl_file = client.evaluators.upsert(**file_dict)  # type: ignore [arg-type, assignment]
-
```

Comment:
Also is removing this a breaking change?

Reply:
more of some hanging code, we do not support evals.run on evaluators